src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3936,15 +3936,17 @@ private void AddDisposingFinally(IOperation resource, bool requiresRuntimeConver
 
             IOperation? tryDispose(IOperation value)
             {
-                Debug.Assert(disposeMethod is object || value.Type!.Equals(iDisposable));
+                Debug.Assert((disposeMethod is object && !disposeArguments.IsDefault) || value.Type!.Equals(iDisposable));
```

Comment:
>value.Type!.Equals(iDisposable) [](start = 89, length = 31)

It feels like we should be able to assert something about ```disposeArguments``` on this side as well.